Build failure: aws-lambda-builders

### Steps To Reproduce

Steps to reproduce the behavior: `nix-build -E 'with import (builtins.fetchTarball { url = "https://github.com/nixos/nixpkgs/archive/d4660b45bc031d98cecd233babf1df7f6d64b65e.tar.gz"; sha256 = "1x3lnwjwlv6cj7jk8zv63h0jpr2iwqh7gifmzmzbiks7nf1v820n"; }) {}; python3.withPackages (p: [p.aws-lambda-builders])'`

### Build log

Here's there 3 test failures:
```
=========================== short test summary info ============================
FAILED tests/functional/workflows/python_pip/test_packager.py::TestSdistMetadataFetcher::test_setup_tar_gz_hyphens_in_name - aws_lambda_builders.workflows.python_pip.packager.UnsupportedPackageError: ...
FAILED tests/functional/workflows/python_pip/test_packager.py::TestSdistMetadataFetcher::test_both_tar_gz - aws_lambda_builders.workflows.python_pip.packager.UnsupportedPackageError: ...
FAILED tests/functional/workflows/python_pip/test_packager.py::TestSdistMetadataFetcher::test_both_tar_bz2 - aws_lambda_builders.workflows.python_pip.packager.UnsupportedPackageError: ...
=========== 3 failed, 615 passed, 286 deselected, 1 warning in 3.36s ===========
/nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 1593: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/qav2qpjrpdw54797mbjxwkj9vrrsbjqp-python3.10-aws-lambda-builders-1.27.0.drv' failed with exit code 1;
       last 10 log lines:
       >   /build/source/tests/functional/test_utils.py:81: DeprecationWarning: Please use assertRaisesRegex instead.
       >     self.assertRaisesRegexp(
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED tests/functional/workflows/python_pip/test_packager.py::TestSdistMetadataFetcher::test_setup_tar_gz_hyphens_in_name - aws_lambda_builders.workflows.python_pip.packager.UnsupportedPackageError: ...
       > FAILED tests/functional/workflows/python_pip/test_packager.py::TestSdistMetadataFetcher::test_both_tar_gz - aws_lambda_builders.workflows.python_pip.packager.UnsupportedPackageError: ...
       > FAILED tests/functional/workflows/python_pip/test_packager.py::TestSdistMetadataFetcher::test_both_tar_bz2 - aws_lambda_builders.workflows.python_pip.packager.UnsupportedPackageError: ...
       > =========== 3 failed, 615 passed, 286 deselected, 1 warning in 3.36s ===========
       > /nix/store/pw17yc3mwmsci4jygwalj8ppg0drz31v-stdenv-linux/setup: line 1593: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/qav2qpjrpdw54797mbjxwkj9vrrsbjqp-python3.10-aws-lambda-builders-1.27.0.drv'.
error: 1 dependencies of derivation '/nix/store/2j2p02gpjhd8yf33j2ya4n7kdrqml50w-python3-3.10.10-env.drv' failed to build

```

### Additional context

I bisected and found that this has been broken ever since we upgraded from setuptools 65 to 67 in https://github.com/NixOS/nixpkgs/commit/451c63214766d3bb8d078620cff4dc74463c5bc8. I don't think this is nixpkgs's fault, I filed this issue upstream to request support for the latest version of setuptools: https://github.com/aws/aws-lambda-builders/issues/474

### Notify maintainers

@lo1tuma FYI

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.8, NixOS, 23.05 (Stoat), 23.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/store/ywgqrgxfs5z0r40c65k0pn9iicj1h55g-yrm0pxss7blbv6fnbq0i30dpjzlcnmz5-source`

```

--- a/pkgs/development/python-modules/aws-lambda-builders/default.nix
+++ b/pkgs/development/python-modules/aws-lambda-builders/default.nix
@@ -2,6 +2,7 @@
 , lib
 , buildPythonPackage
 , fetchFromGitHub
+, fetchpatch
 , mock
 , parameterized
 , pyelftools
@@ -26,6 +27,15 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [
     six
+  ];
+
+  patches = [
+    # This patch can be removed once https://github.com/aws/aws-lambda-builders/pull/475 has been merged.
+    (fetchpatch {
+      name = "setuptools-66-support";
+      url = "https://patch-diff.githubusercontent.com/raw/aws/aws-lambda-builders/pull/475.patch";
+      sha256 = "sha256-EkYQ6DNzbSnvkOads0GFwpGzeuBoLVU42THlSZNOHMc=";
+    })
   ];
 
   nativeCheckInputs = [

